cpu/stm32/stmclk/stmclk_l4wb.c

```diff
@@ -151,50 +222,86 @@ void stmclk_init_sysclk(void)
              (instead of MSIRANGE in the RCC_CR) */
     RCC->CR = (RCC_CR_HSION);
 
-#if (CLOCK_HSE)
-    /* if configured, we need to enable the HSE clock now */
-    RCC->CR |= (RCC_CR_HSEON);
-    while (!(RCC->CR & RCC_CR_HSERDY)) {}
-#endif
+    /* Enable the HSE clock only when it's provided by the board and required:
+        - Use HSE as system clock
+        - Use HSE as PLL input clock
+    */
+    if (IS_ACTIVE(CONFIG_BOARD_HAS_HSE) &&
+        (IS_ACTIVE(CONFIG_USE_CLOCK_HSE) || IS_ACTIVE(CONFIG_CLOCK_PLL_SRC_HSE))) {
+        RCC->CR |= (RCC_CR_HSEON);
+        while (!(RCC->CR & RCC_CR_HSERDY)) {}
+    }
 
-#if ((CLOCK_HSE == 0) || CLOCK_MSI_ENABLE)
-    /* reset clock to MSI with 48MHz, disables all other clocks */
+    if (IS_ACTIVE(CONFIG_USE_CLOCK_HSE)) {
+        /* Select HSE as system clock and configure the different prescalers */
+        RCC->CFGR &= ~RCC_CFGR_SW;
+        RCC->CFGR |= RCC_CFGR_SW_HSE;
+    }
+    else if (IS_ACTIVE(CONFIG_USE_CLOCK_MSI)) {
 #if defined(CPU_FAM_STM32WB)
```

Comment:
Can `IS_ACTIVE`be used here?

Reply:
same as https://github.com/RIOT-OS/RIOT/pull/14866#discussion_r484874097

Reply:
same as https://github.com/RIOT-OS/RIOT/pull/14866#discussion_r484874097